internal/eventstream/eventstream.go

```diff
@@ -85,7 +85,10 @@ func (i *inMemoryEventStream[T, E]) Changes(ctx context.Context) (iter.Seq[T], e
 }
 
 func (i *inMemoryEventStream[T, E]) View(ctx context.Context) (T, error) {
-	return i.view, nil
+	i.lock.Lock()
```

Comment:
`Publish()` safely writes to `i.view` with a lock while `View()` read without a lock